cfme/tests/v2v/test_v2v_migrations_single_vcenter.py

```diff
@@ -284,3 +286,56 @@ def _cleanup():
 
     migrated_vm = get_migrated_vm(src_vm_obj, provider)
     assert src_vm_obj.mac_address == migrated_vm.mac_address
+
+
+@pytest.mark.tier(4)
+def test_migration_restart(appliance, provider, mapping_data_vm_obj_mini):
+    """
+    Test migration by restarting evmserverd in middle of the process
+
+    Polarion:
+        assignee: ytale
+        initialEstimate: 1h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.10
+        casecomponent: V2V
+    """
+    src_vm_obj = mapping_data_vm_obj_mini.vm_list[0]
+    migration_plan_collection = appliance.collections.v2v_migration_plans
+    migration_plan = migration_plan_collection.create(
+        name="plan_{}".format(fauxfactory.gen_alphanumeric()),
+        description="desc_{}".format(fauxfactory.gen_alphanumeric()),
+        infra_map=mapping_data_vm_obj_mini.infra_mapping_data.get("name"),
+        target_provider=provider,
+        vm_list=mapping_data_vm_obj_mini.vm_list
+    )
+    view = navigate_to(migration_plan, "InProgress")
+    assert migration_plan.wait_for_state("Started")
+
+    def _system_reboot():
+        # reboot system when migrated percentage greater than 20%
+        ds_percent = int(view.progress_card.get_progress_percent(migration_plan.name)["datastores"])
+        if ds_percent > 20:
+            appliance.restart_evm_rude()
+            appliance.wait_for_web_ui()
```

Comment:
I don't think its a good pattern to wait_for_web_ui here, as you're already in a wait context.

This should call restart_evm_rude(), and then return `True` to stop the wait_for around this method.

Then call `appliance.wait_for_web_ui` after the wait_for at L328 completes.

Reply:
Totally make sense, updated, thanks.

Reply:
@Yadnyawalkya You've added wait_for_web_ui on L344 below, but didn't remove it here.